packages/shared/lib/services/activity/activity.service.ts

```diff
@@ -335,3 +335,27 @@ export async function createActivityLogDatabaseErrorMessageAndEnd(baseMessage: s
         content: errorMessage
     });
 }
+
+export async function findOldActivities({ retention, limit }: { retention: number; limit: number }): Promise<{ id: number }[]> {
+    const q = db.knex
+        .queryBuilder()
+        .withSchema(db.schema())
+        .from('_nango_activity_logs')
+        .select('id')
+        .where(db.knex.raw(`_nango_activity_logs.updated_at <  NOW() - INTERVAL '${retention} days'`))
+        .limit(limit);
+    const logs: { id: number }[] = await q;
+
+    return logs;
+}
+
+export async function deleteLog({ activityLogId }: { activityLogId: number }): Promise<void> {
+    await db.knex.withSchema(db.schema()).from('_nango_activity_logs').where({ id: activityLogId }).del();
+}
+
+export async function deleteLogsMessages({ activityLogId, limit }: { activityLogId: number; limit: number }): Promise<number> {
+    const del = await db.knex.raw(
```

Comment:
Why is this raw? No way to keep this with a knee call?

Reply:
Indeed we can, I thought it wasn't possible to where in select